connectors/sources/sharepoint_online.py

```diff
@@ -619,6 +623,16 @@ async def users_and_groups_for_role_assignment(self, site_web_url, role_assignme
             # This can also mean "not found" so handling it explicitly
             return []
 
+    async def groups_for_user(self, site_web_url, user_id):
+        self._validate_sharepoint_rest_url(site_web_url)
+
+        url = f"{site_web_url}/_api/web/GetUserById('{user_id}')/groups"
+
+        try:
+            return await self._rest_api_client.fetch(url)
```

Comment:
This one doesn't need to be `scroll`ed?

Reply:
AFAIU not necessarily as this is specific to the user and not the pages, but I may misunderstood the topology of SPO.